### ISSUE TITLE
run lightning run app app.py --cloud from lightning notebook

### ISSUE THREAD

**Issue by robert-s-lee** [Lightning-AI/lightning#13864]
## 🚀 Feature

run lightning run app app.py --cloud from a lightning notebook

### Motivation

need to access bigger CPU and GPU instances when developing a lightning app.  

the local laptop environment does not provide sufficient resources when developing a complex app.   more info at https://pytorch-lightning.slack.com/archives/CRBLFHY79/p1658770802365889

### Pitch

Use Lightning [Notebook app](https://lightning.ai/app/hvUwbEG70BScratchPad%2C%20Notebook%20Manager%20for%20Teams) as my laptop in the cloud.  

running `lightning run app app.py --cloud --name quick-start-workshop`  results in the following:

- waits on this message
`Your Lightning App is starting. This won't take long.`
- ctl-c and received this message
`AttributeError: Authentication Failed, no authentication header available.`  

- tried `lightning login`
```
lightning login
INFO: login started for lightning.ai, opening https://lightning.ai/sign-in?redirectTo=http%3A%2F%2Flocalhost%3A56207%2Flogin-complete
```

- tried copying tried copying .lightning/crednential.json file from my laptop and putting in /home/zeus and resulted in the above issues

**Comment by hhsecond** [comment]
@robert-s-lee Thanks for the ticket. The login mechanism currently requires us to have UI opened  which isn’t great. As a workaround, you can either copy the credential file from your local machine to `/home/zeus/.lightning/` inside your notebook machine or set the credentials using the env variables `LIGHTNING_USER_ID`  and `LIGHTNING_API_KEY`.

We should ideally fix the login flow but about your current bug reported, I wasn’t able to reproduce. Could you verify if your credential file is exactly as same as what you have in local and and the location is right

```bash
$ pwd
/home/zeus/.lightning
$ ls
cache  credentials.json
$ 
```

**Comment by robert-s-lee** [comment]
ran
```
lightning run app app.py --cloud --open-ui false
```

received the following

```
  File "/content/venv/lib/python3.8/site-packages/lightning_app/utilities/packaging/lightning_utils.py", line 107, in _prepare_lightning_wheels_and_requirements
    tar_name = _copy_tar(_PROJECT_ROOT, root)
  File "/content/venv/lib/python3.8/site-packages/lightning_app/utilities/packaging/lightning_utils.py", line 73, in _copy_tar
    tar_files = os.listdir(dist_dir)
FileNotFoundError: [Errno 2] No such file or directory: '/content/venv/lib/python3.8/dist'
```

**Comment by hhsecond** [comment]
that’s odd and looks like a regression on the packaging function (could you create another issue please?). As a work around, can you set the env variable `PACKAGE_LIGHTNING` to `0` and try?

**Comment by robert-s-lee** [comment]
@hhsecond I got the a success run from the cloud.  the issue was https://github.com/Lightning-AI/lightning/issues/13942.  The steps to make this work:

```
cd
mkdir ~/.lightning
cat  ~/.lightning/credentials.json <<EOF
copy credentials from laptop
EOF
```

the output here.

```
zeus@lightningwork-01g9an3ze5h1z2x4ksrwyfgj7v:~/test$ lightning run app app.py --cloud --open-ui false
Your Lightning App is starting. This won't take long.
The name of the app is: singing-tesla-4089
upload ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 100.0%
Application is ready in the cloud
```
